### PR TITLE
when transaction should not notify

### PR DESCRIPTION
**Pull request by jiqimaogou**
the data in database is old when in transaction. so it should not notify.

### CODE REVIEW

File: library/src/main/java/com/raizlabs/android/dbflow/runtime/TransactionManager.java [hunk 1]
```diff
@@ -124,6 +125,24 @@ public static void transact(SQLiteDatabase database, Runnable runnable) {
...
+     */
+    public static void transact(Class<? extends Model> table, Runnable runnable) {
+        SQLiteDatabase database = FlowManager.getDatabaseForTable(table).getWritableDatabase();
+        database.beginTransaction();
+        try {
+            runnable.run();
+            database.setTransactionSuccessful();
+        } finally {
+            database.endTransaction();
+            SqlUtils.notifyModelChanged(table, null);
```
Comment (agrosner) [inline library/src/main/java/com/raizlabs/android/dbflow/runtime/TransactionManager.java]: why did we add this here? this is not the intention of the method.

Comment (agrosner) [inline library/src/main/java/com/raizlabs/android/dbflow/runtime/TransactionManager.java]: #req please remove

File: library/src/main/java/com/raizlabs/android/dbflow/sql/SqlUtils.java [hunk 2]
```diff
@@ -409,7 +409,10 @@ void delete(final TableClass model, AdapterClass modelAdapter, boolean async) {
      * @param table  The table of the model
      */
     public static void notifyModelChanged(Class<? extends Model> table, BaseModel.Action action) {
-        FlowManager.getContext().getContentResolver().notifyChange(getNotificationUri(table, action), null, true);
+        SQLiteDatabase db = FlowManager.getDatabaseForTable(table).getWritableDatabase();
+        if (!db.inTransaction()) {
```
Comment (agrosner) [inline library/src/main/java/com/raizlabs/android/dbflow/sql/SqlUtils.java]: why do we care if the database is in a transaction or not, to notify a model changed?

Comment (jiqimaogou) [inline library/src/main/java/com/raizlabs/android/dbflow/sql/SqlUtils.java]: the transaction is commit after we call ```database.endTransaction();```, then the change is commit to the database.

if we query a model in transaction, the data in database is still old, we get the old models, we always want to get the new data when is notified, but the notify in transaction make us query the old model again.

```
        try {
           model.save();
           new Select().from(model.getClass()).query(); // the data is still old
            database.setTransactionSuccessful();
        } finally {
            database.endTransaction();
            SqlUtils.notifyModelChanged(table, null);
        }
```

Comment (agrosner) [inline library/src/main/java/com/raizlabs/android/dbflow/sql/SqlUtils.java]: Ok So this became a lot more _interesting_ of an issue. i will have to introduce in `2.0.0` a new way to notify when models change outside of transactions.

**Comment by agrosner** [comment]
closing PR

**Comment by brucexia0** [comment]
I Don't think observer.beginTransaction;observer.EndTransactionAndNotify is the answer. Typically in the code that commits database changes wouldn't have access to observers. Why not simply quell the notification and only fire them after transaction is commited? @agrosner

**Pull request closed without merging**